kirin/command/piv_worker.py

```diff
@@ -173,8 +173,10 @@ def piv_worker():
                 should_wait = False  # wait only after init crash
                 logger.info("launching the PIV worker for '{0}'".format(contributor.id))
                 worker.run()
+            db.session.commit()
         except Exception as e:
             logger.warning("PIV worker died: {0}".format(e))
+            db.session.rollback()
         finally:
```

Comment:
How about inserting here:
```suggestion
        finally:
            try:
                db.session.commit()
            except Exception as db_e:
                logger.warning("Exception while db-commit: {0}".format(db_e))
                db.session.rollback()
```
The idea would be to avoid direct rollback when another exception bring us in the except.